jsapp/js/components/modalForms/connectProjects.es6

```diff
@@ -0,0 +1,367 @@
+        labels: {ok: t('Acknowledge and continue'), cancel: t('Cancel')},
+        onok: (evt, value) => {
+          actions.dataShare.toggleDataSharing(this.props.asset.uid, data);
+          dialog.destroy();
+        },
+        oncancel: () => {
+          dialog.destroy();
+        }
+      };
+      dialog.set(opts).show();
+    } else {
+      actions.dataShare.toggleDataSharing(this.props.asset.uid, data);
+    }
+  }
+
+  /*
+   * Utilities
+   */
+
+  generateAutoname(newParent) {
+    if (newParent) {
+      let autoname = newParent.name;
+      autoname = autoname.toLowerCase().substring(0, 30).replace(/(\ |\.)/g, '_');
+      this.setState({newFilename: autoname});
+    }
+  }
+  generateTruncatedDisplayName(name) {
+    return name.length > 30 ? `${name.substring(0, 30)}...` : name;
+  }
+  generateFilteredAssetList() {
+    let attachedParentUids = [];
+    this.state.attachedParents.forEach((item) => {
+      attachedParentUids.push(item.parentUid)
+    });
+
+    // Filter displayed asset list based on unattached projects
+    return (
+      this.state.sharingEnabledAssets.results.filter(
+        item => !attachedParentUids.includes(item.uid)
+      )
+    );
+  }
+
+ /*
+  * May be useful later for replacing autoname with existing name
+  *
+  * getExteralFilename() {
+  *   let filename = '';
+  *   this.props.asset.content.survey.some((element) => {
+  *     if (element.type === XML_EXTERNAL) {
+  *       filename = element.name;
+  *     }
+  *   });
+  *   return filename;
+  * }
+  */
+
+  /*
+   * Rendering
+   */
+
+  renderLoading(message = t('loading…')) {
+    return (
+      <bem.Loading>
+        <bem.Loading__inner>
+          <i />
+          {message}
+        </bem.Loading__inner>
+      </bem.Loading>
+    );
+  }
+  renderSwitchLabel() {
+    if (this.state.isShared) {
+      return (
+        <ToggleSwitch
+          onChange={this.toggleSharingData.bind(this)}
+          label={t('Data sharing enabled')}
+          checked={this.state.isShared}
+        />
+      );
+    } else {
+      return (
+        <ToggleSwitch
+          onChange={this.toggleSharingData.bind(this)}
+          label={t('Data sharing disabled')}
+          checked={this.state.isShared}
+        />
+      );
+    }
+  }
+  renderSelect(sharingEnabledAssets) {
+    const selectClassNames = ['kobo-select__wrapper'];
+    if (this.state.fieldsErrors.emptyParent || this.state.fieldsErrors.parent) {
+      selectClassNames.push('kobo-select__wrapper--error');
+    }
+    return(
+      <div className={selectClassNames.join(' ')}>
+        <Select
+          placeholder={t('Select a different project to import data from')}
+          options={sharingEnabledAssets}
+          value={this.state.newParent}
+          isLoading={(this.state.isVirgin || this.state.isLoading || !sharingEnabledAssets)}
```

Comment:
` || !sharingEnabledAssets` would never happen, right? As this `Select` never renders when `sharingEnabledAssets` is empty (i.e. false)